test/e2e/storage/flexvolume_resize.go

```diff
@@ -0,0 +1,169 @@
+		Expect(*resizableSc.AllowVolumeExpansion).To(BeTrue())
+
+		pvc = newClaim(test, ns, "default")
+		pvc.Spec.StorageClassName = &resizableSc.Name
+		pvc, err = c.CoreV1().PersistentVolumeClaims(pvc.Namespace).Create(pvc)
+		Expect(err).NotTo(HaveOccurred(), "Error creating pvc: %v", err)
+
+	})
+
+	framework.AddCleanupAction(func() {
+		if len(nodeLabelValue) > 0 {
+			framework.RemoveLabelOffNode(c, nodeName, nodeKey)
+		}
+	})
+
+	AfterEach(func() {
+		framework.Logf("AfterEach: Cleaning up resources for mounted volume resize")
+
+		if c != nil {
+			if errs := framework.PVPVCCleanup(c, ns, nil, pvc); len(errs) > 0 {
+				framework.Failf("AfterEach: Failed to delete PVC and/or PV. Errors: %v", utilerrors.NewAggregate(errs))
+			}
+			pvc, nodeName, isNodeLabeled, nodeLabelValue = nil, "", false, ""
+			nodeKeyValueLabel = make(map[string]string)
+		}
+	})
+
+	It("should be resizable when mounted", func() {
+		driver := "dummy-attachable"
+
+		node := nodeList.Items[0]
+		By(fmt.Sprintf("installing flexvolume %s on node %s as %s", path.Join(driverDir, driver), node.Name, driver))
+		installFlex(c, &node, "k8s", driver, path.Join(driverDir, driver))
+
+		pv := framework.MakePersistentVolume(framework.PersistentVolumeConfig{
+			PVSource: v1.PersistentVolumeSource{
+				FlexVolume: &v1.FlexPersistentVolumeSource{
+					Driver: "k8s/" + driver,
+				}},
+			NamePrefix:       "pv-",
+			StorageClassName: resizableSc.Name,
+			VolumeMode:       pvc.Spec.VolumeMode,
+		})
+
+		pv, err = framework.CreatePV(c, pv)
+		Expect(err).NotTo(HaveOccurred(), "Error creating pv %v", err)
+
+		By("Waiting for PVC to be in bound phase")
+		pvcClaims := []*v1.PersistentVolumeClaim{pvc}
+		var pvs []*v1.PersistentVolume
+
+		pvs, err = framework.WaitForPVClaimBoundPhase(c, pvcClaims, framework.ClaimProvisionTimeout)
+		Expect(err).NotTo(HaveOccurred(), "Failed waiting for PVC to be bound %v", err)
+		Expect(len(pvs)).To(Equal(1))
+
+		var pod *v1.Pod
+		By("Creating pod")
+		pod, err = framework.CreateNginxPod(c, ns, nodeKeyValueLabel, pvcClaims)
+		Expect(err).NotTo(HaveOccurred(), "Failed to create pod %v", err)
+		defer framework.DeletePodWithWait(f, c, pod)
+
+		By("Waiting for pod to go to 'running' state")
+		err = f.WaitForPodRunning(pod.ObjectMeta.Name)
+		Expect(err).NotTo(HaveOccurred(), "Pod didn't go to 'running' state %v", err)
+
+		By("Expanding current pvc")
+		newSize := resource.MustParse("6Gi")
+		pvc, err = expandPVCSize(pvc, newSize, c)
+		Expect(err).NotTo(HaveOccurred(), "While updating pvc for more size")
+		Expect(pvc).NotTo(BeNil())
+
+		pvcSize := pvc.Spec.Resources.Requests[v1.ResourceStorage]
+		if pvcSize.Cmp(newSize) != 0 {
+			framework.Failf("error updating pvc size %q", pvc.Name)
+		}
+
+		By("Waiting for cloudprovider resize to finish")
+		err = waitForControllerVolumeResize(pvc, c)
+		Expect(err).NotTo(HaveOccurred(), "While waiting for pvc resize to finish")
+
+		By("Waiting for file system resize to finish")
```

Comment:
I thought fs resize by default will only happen when pod is restarted. How come in this case volume is being resized without waiting for it? The online expansion feature is still in Alpha.

Reply:
@gnufied this test is for online resize. So perhaps I can add a Skip unless `-feature-gates=ExpandPersistentVolumes=true,ExpandInUsePersistentVolumes=true` is passed to the test.

Reply:
So now that we don't depend on ONLINE resizing, don't we need to restart the pod for flex volume?

Reply:
> So now that we don't depend on ONLINE resizing, don't we need to restart the pod for flex volume?

1. basically if the online expansion feature gate is ON, then the resizing should be done w/o pod restarts
2. else it should be _after_ a pod restarts.

For #2 tests already exist. For #1 I'm adding the new test in this PR that's why its not expecting the pod to restart.